pkg/power/ethtool.go

```diff
@@ -0,0 +1,69 @@
+package power
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type Flag string
+
+const (
+	SupportsWakeOn Flag = "Supports Wake-on"
+	WakeOn         Flag = "Wake-on"
+)
+
+type WolMode string
+
+const (
+	MagicPacket WolMode = "g"
+)
+
+var (
+	ErrFlagNotFound = fmt.Errorf("flag not found")
+)
+
+func ethtool(ctx context.Context, arg ...string) ([]byte, error) {
+	return exec.CommandContext(ctx, "ethtool", arg...).CombinedOutput()
+}
+
+func ValueOfFlag(ctx context.Context, nic string, flag Flag) (string, error) {
+	output, err := ethtool(ctx, nic)
+	if err != nil {
+		return "", err
+	}
+
+	return valueOfFlag(output, flag)
+}
+
+func valueOfFlag(output []byte, flag Flag) (string, error) {
+	buf := bytes.NewBuffer(output)
+	scanner := bufio.NewScanner(buf)
+	for scanner.Scan() {
+		if err := scanner.Err(); err != nil {
+			return "", err
+		}
+
+		line := strings.TrimSpace(scanner.Text())
+		parts := strings.Split(line, ":")
+		if parts[0] != string(flag) {
+			continue
+		}
+		if len(parts) != 2 {
+			return "", fmt.Errorf("invalid ethtool output format")
+		}
+		return strings.TrimSpace(string(parts[1])), nil
+	}
+
+	return "", ErrFlagNotFound
```

Comment:
I think this error should be dynamic to include the flag that was requested in the error message?

Reply:
I checked but this is returned when u are trying to get the value of `specific` flag. So when flag not found is returned it means this specific flag u asking for is not found. So when this error is returned, no extra information need to be attached to the error since the caller already know what it was asking